### DISCUSSION TITLE
I built a browser extension to make exporting your subscriptions much easier

### DISCUSSION THREAD

**Discussion by jakenvac** [FreeTubeApp/FreeTube#2369] [General]
Hi, just thought I'd share this here, as I will on the newpipe repo too.

I built a browser extension to export your subscriptions in the google takeout csv format. I have tested it on firefox and successfully imported about 250 subscriptions into freetube.

You can find it [here](https://github.com/jakehl/ytsubgrabber). Be sure to read the readme regarding installing it.

Disclaimer: I am not currently working on it any more. It has served it's purpose for me and unfortunately I just don't have the time to maintain it. I have released the code into the public domain.

The reason I thought to share it is that someone else may want to pick it up and develop it further, or possibly it could be taken over by the freetube or Newpipe team. 

Here are some of the goals I had for the extension that never got implemented:
- Add an 'Open in freetube/newpipe' button underneath a video/on context menus etc. (replace https:// with freetube:// or a newpipe scheme if there is one)
- Create a native messenger to allow playlists on the youtube site to use freetube as the video player (not even sure this is possible, highly doubt it is for a newpipe version)
- Export history, playlists etc.

**Comment by PrestonN** [comment]
Hey there! That's a neat little app you made.

While the app seems useful, I don't really see us taking over any ownership of it any time soon.

I'd imagine it isn't very common for users to try to have a subscription list on both YouTube and FreeTube. I understand that isn't always the case, the fact you made a tool to help with it is proof of that. Trying to support essentially syncing FreeTube subscriptions up with YouTube isn't a workflow that we'll want to maintain however. I'd much rather look into more independent methods of syncing up subscriptions across devices / instances.

As for your planned features, we do intend on allowing importing of history and playlists from Google Takeout, though it'll likely be a little more manual similar to how subscriptions currently work. The other ideas you had sounds like something that might go well with another project like Privacy Redirect / LibRedirect. I would maybe consider working with those teams to see if those features could be implemented.

Regardless, new tools is always a benefit to everyone, so thanks for creating something that others might have a use for. Even if you're not maintaining it, I'm sure others will have a use for it.
